gson/src/test/java/com/google/gson/functional/NamingPolicyTest.java

```diff
@@ -237,6 +238,78 @@ public void testAtSignInSerializedName() {
     assertThat(new Gson().toJson(new AtName())).isEqualTo("{\"@foo\":\"bar\"}");
   }
 
+  @Test
+  public void testGsonWithNameDeserialiation() {
+    Gson gson =
+        builder
+            .setFieldNamingStrategy(
+                new FieldNamingStrategy() {
+
+                  @Override
+                  public String translateName(Field f) {
+                    return "SomeConstantStringInstanceField";
+                  }
+
+                  @Override
+                  public List<String> alternateNames(Field f) {
+                    return List.of("alternate-name");
+                  }
+                })
+            .create();
+    String target = "{\"SomeConstantStringInstanceField\":\"someValue\"}";
+    StringWrapper deserializedObject = gson.fromJson(target, StringWrapper.class);
+    assertThat(deserializedObject.someConstantStringInstanceField).isEqualTo("someValue");
+  }
+
+  @Test
+  public void testGsonWithAlternateNamesDeserialiation() {
+    Gson gson =
+        builder
+            .setFieldNamingStrategy(
+                new FieldNamingStrategy() {
+
+                  @Override
+                  public String translateName(Field f) {
+                    return "primary-name";
+                  }
+
+                  @Override
+                  public List<String> alternateNames(Field f) {
+                    return List.of("SomeConstantStringInstanceField");
+                  }
+                })
+            .create();
+    String target = "{\"SomeConstantStringInstanceField\":\"someValue\"}";
```

Comment:
Could you please change this name here to a more custom name? I am a bit worried that `SomeConstantStringInstanceField` is too close to the real field name. So it could look like the field naming strategy is not actually used.
```suggestion
                    return List.of("alternate-name");
                  }
                })
            .create();
    String target = "{\"alternate-name\":\"someValue\"}";
```